src/distro/zulu.ts

```diff
@@ -0,0 +1,150 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+import * as httpm from '@actions/http-client';
+import * as semver from 'semver';
+
+const IS_WINDOWS = process.platform === 'win32';
+
+export async function getJavaZulu(version: string, javaPackage: string) {
+  core.debug('Downloading JDK from Azul');
+  const http = new httpm.HttpClient('setup-java', undefined, {
+    allowRetries: true,
+    maxRetries: 3
+  });
```

Comment:
May be we could use some inheritance here.
We can create base class `JavaInstaller` and two child classes: `AzulInstaller`, `AdoptInstaller`. It would allow us to move some common logic to base class and avoid code duplication.
For example, http initialization could be put to constructor and re-used.